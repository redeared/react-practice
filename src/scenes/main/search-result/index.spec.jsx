import React from 'react';
import { shallow } from 'enzyme';

import { create } from 'react-test-renderer';
import { SearchResult } from './index';

describe('SearchResult', () => {
    it('should render without crash', () => {
        const comp = create(<SearchResult
            items={[{ title: 'title' }]}
            onItemClick={() => 'nothing'}
        />).toJSON();
        expect(comp).toMatchSnapshot();
    });

    it('should update filters on updateFilters', () => {
        const comp = shallow(<SearchResult />);
        comp.setState({
            ...comp.state(), sortByRating: false, sortByReleaseDate: false,
        });
        comp.instance().updateFilters({ sortByRating: true, sortByReleaseDate: true });

        expect(comp.state().sortByRating).toBe(true);
        expect(comp.state().sortByReleaseDate).toBe(true);
    });

    it('should sort items first by rating, second by Date', () => {
        const first = { date: 2000, rating: 10 };
        const second = { date: 2001, rating: 8 };
        const third = { date: 2000, rating: 8 };
        const fourth = { date: 2001, rating: 7 };

        const items = [fourth, third, second, first];

        const comp = shallow(<SearchResult items={items} />);
        comp.setState({
            ...comp.state(), sortByRating: true, sortByReleaseDate: true,
        });

        expect(comp.instance().getSortedItems()[0]).toBe(first);
        expect(comp.instance().getSortedItems()[1]).toBe(second);
        expect(comp.instance().getSortedItems()[2]).toBe(third);
        expect(comp.instance().getSortedItems()[3]).toBe(fourth);
    });

    it('should return [] by getItems if items input param is not array ', () => {
        const comp = shallow(<SearchResult items={{}} />);
        expect(comp.instance().getItems()).toEqual([]);
    });
});
