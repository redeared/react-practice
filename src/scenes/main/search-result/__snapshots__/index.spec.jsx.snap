// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult should render without crash 1`] = `
Array [
  <div
    style={
      Object {
        "background": "ghostwhite",
        "padding": "0 4em 0 4em",
      }
    }
  >
    <span>
       
      1
       movie
       found 
    </span>
    <span
      style={
        Object {
          "float": "right",
        }
      }
    >
      Sort by:
       
      <span
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={null}
        tabIndex="0"
      >
        release date
      </span>
       
      <span
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "color": "#f45362",
          }
        }
        tabIndex="0"
      >
        rating
      </span>
    </span>
  </div>,
  <div
    style={
      Object {
        "background": "white",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": "100%",
        "overflowY": "auto",
      }
    }
  >
    <div
      onClick={[Function]}
      role="presentation"
      style={
        Object {
          "flexBasis": "25%",
        }
      }
    >
      <img
        alt="title"
        className="img"
        src=""
      />
      <div>
        TITLE
          
        <span />
      </div>
      <div />
    </div>
  </div>,
]
`;
